pyvista/core/_typing_core/_aliases.py

```diff
@@ -18,3 +18,5 @@
 
 TransformLike = Union[Matrix[float], vtkMatrix3x3, vtkMatrix4x4, vtkTransform]
 BoundsLike = Tuple[Number, Number, Number, Number, Number, Number]
+CellsLike = Union[Matrix[int], Vector[int]]
+CellArrayLike = Union[CellsLike, vtkCellArray]
```

Comment:
This is perhaps a bit counter-intuitive, but I think the TypeVar needs to be explicitly specified here. Otherwise, the type will default to `Any` as in
`CellArrayLike = Union[CellsLike[Any], vtkCellArray]`
```suggestion
CellArrayLike = Union[CellsLike[int], vtkCellArray]
```

Reply:
*what?!* 

actually if I write `check_cellslike.py`:

```python
from typing import reveal_type
from core._typing_core import CellsLike

x: CellsLike = [3, 0, 1, 2]
reveal_type(x)

y: CellsLike[int] = [3, 0, 1, 2]
reveal_type(y)
```

and mypy 1.7.1 says:
```
check_cellslike.py:6: note: Revealed type is "Any"
check_cellslike.py:9: note: Revealed type is "Any"
```

and now i'm even more confused. trying to reduce the issue, `check_cellslike_minimal.py`:

```python
from typing import reveal_type, TypeVar, List, Union

T = TypeVar('T')
Vector = List[T]
Matrix = List[List[T]]
CellsLike = Union[Vector[int], Matrix[int]]

x: CellsLike = [3, 0, 1, 2]
reveal_type(x)

y: CellsLike[int] = [3, 0, 1, 2]
reveal_type(y)
```

gives the (expected):

```
check_cellslike_minimal.py:9: note: Revealed type is "Union[builtins.list[builtins.int], builtins.list[builtins.list[builtins.int]]]"
check_cellslike_minimal.py:11: error: Bad number of arguments for type alias, expected: 0, given: 1  [type-arg]
check_cellslike_minimal.py:12: note: Revealed type is "Union[builtins.list[builtins.int], builtins.list[builtins.list[builtins.int]]]"
```

is this related to the custom mypy plugin from #5285? Or some other mypy black magic?

Reply:
My mistake, I believe the implementation as-is is correct since  `Matrix` in `CellsLike` is bound  to `int` explicitly, and not to a TypeVar.  No need to make any of the suggested changes.

The concern I had would only apply in the case where `CellsLike` would be generic, e.g. (slightly modified version of your second example):

``` python
from typing import reveal_type, TypeVar, List, Union

T = TypeVar('T')
Vector = List[T]
Matrix = List[List[T]]
IntType = TypeVar('IntType', bound=int) 
CellsLike = Union[Vector[IntType], Matrix[IntType]]

x: CellsLike = [3, 0, 1, 2]
reveal_type(x)

y: CellsLike[int] = [3, 0, 1, 2]
reveal_type(y)
```

which yields the following, with an assumed `Any` type.
```
<string>:10: note: Revealed type is "Union[builtins.list[Any], builtins.list[builtins.list[Any]]]"
<string>:13: note: Revealed type is "Union[builtins.list[builtins.int], builtins.list[builtins.list[builtins.int]]]"
```

For your first example, how are you calling `mypy`? It could be that mypy can't find pyvista, so it doesn't know the type, and therefore spits out `Any`. E.g. doing:
``` bash
mypy -c 'from pyvista.core._typing_core import Vector; x: Vector[int]; reveal_type(x)' --follow-imports skip
```
will yield
``` python
note: Revealed type is "Any"
```
because of the `--follow-imports skip` flag. But running without the flag will yield the expected result.